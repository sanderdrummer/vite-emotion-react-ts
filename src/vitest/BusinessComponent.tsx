import { useState } from "react";

export const MockMeComponent = ({ handleClick = () => {} }) => {
  return <button onClick={handleClick}>click me</button>;
};

export const BusinessComponent = () => {
  const [show, setShow] = useState(false);
  return (
    <>
      <MockMeComponent handleClick={() => setShow(!show)} />
      {show && <div>yeah vite</div>}
    </>
  );
};
