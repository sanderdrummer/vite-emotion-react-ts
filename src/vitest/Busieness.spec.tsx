import { render, screen, fireEvent } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";
import { BusinessComponent, MockMeComponent } from "./BusinessComponent";

describe("BusinessComponent", () => {
  it("toggles yeah vite", () => {
    render(<BusinessComponent />);
    const button = screen.getByRole("button", { name: "click me" });
    expect(screen.queryByText("yeah vite")).not.toBeInTheDocument();

    fireEvent.click(button);
    expect(screen.getByText("yeah vite")).toBeInTheDocument();
  });

  it("maches snapshot", () => {
    const { container } = render(<BusinessComponent />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe("MockMeComponent", () => {
  afterEach(() => {
    vi.resetAllMocks();
  });

  it("calls handler", () => {
    const mock = vi.fn();
    render(<MockMeComponent handleClick={mock} />);
    expect(mock).toHaveBeenCalledTimes(0);
    const button = screen.getByRole("button", { name: "click me" });
    fireEvent.click(button);
    expect(mock).toHaveBeenCalledTimes(1);
  });
});
