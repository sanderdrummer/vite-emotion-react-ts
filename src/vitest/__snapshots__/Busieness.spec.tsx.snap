// Vitest Snapshot v1

exports[`BusinessComponent > maches snapshot 1`] = `
<button>
  click me
</button>
`;
